src/ex4/server/services/item_manager.js

```diff
@@ -0,0 +1,89 @@
+const PokemonClient = require("../clients/pokemon_client");
+const { Item } = require("../db/models/item");
+
+class ItemManager {
+  constructor() {
+    this.pokemonClient = new PokemonClient();
+  }
+
+  getItems = () => {
+    try {
+      return Item.findAll();
+    } catch (err) {
+      console.error(err);
+    }
+  };
+
+  handleItem = async (item) => {
+    console.log("item manager - handleItem", item);
+    if (this._isNumber(item)) {
+      return await this.fetchAndAddPokemon(item);
+    }
+    if (this._isList(item)) {
+      return await this.fetchAndAddManyPokemon(item);
+    }
+
+    this.addItem(item);
+  };
+
+  addItem = (item) => {
+    try {
+      console.log("item manager - add item:", { item });
+      Item.create({
+        id: id,
+        ItemName: item,
+        createdAt: new Date(),
+        updatedAt: new Date(),
+      });
+    } catch (err) {
+      console.error(err);
+    }
+  };
+
+  addPokemonItem = async (pokemon) => {
+    try {
+      await Item.create({
+        id: id,
+        ItemName: `Catch ${pokemon.name}`,
+        createdAt: new Date(),
+        updatedAt: new Date(),
+      });
+    } catch (err) {
+      console.error(err);
+    }
+  };
+
+  fetchAndAddPokemon = async (pokemonId) => {
+    try {
+      const pokemon = await this.pokemonClient.getPokemon(pokemonId);
+      this.addPokemonItem(pokemon);
+    } catch (error) {
+      this.addItem(`Pokemon with ID ${pokemonId} was not found`);
+    }
+  };
+
+  fetchAndAddManyPokemon = async (inputValue) => {
```

Comment:
general comment: I see that you're implementing the business logic in the same place that you implemented the api. better split between them to have single responsibility for each method. it will be easier to maintain, as well as onboard additional devs to your project 🎖️